### PR TITLE
Update FPGA tools

### PR DESCRIPTION
**Pull request by emilazy**
###### Motivation for this change
nMigen 0.1 is [approaching release](https://twitter.com/whitequark/status/1183960591825809408), so here's another bump of the FOSS FPGA tools. I've also marked the [Clash](https://clash-lang.org/) Haskell packages as unbroken, since as of the 1.0 release they seem to compile and work fine here.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-hackage2nix.yaml [hunk 1]
```diff
@@ -3623,10 +3623,7 @@ broken-packages:
   - clarifai
   - CLASE
   - clash
-  - clash-ghc
-  - clash-lib
   - clash-multisignal
-  - clash-prelude
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-hackage2nix.yaml]: Could you pull out these changes and send them in a different PR against the `haskel-updates` branch?

Here's a video that describes how to mark Haskell packages as unbroken:

https://discourse.nixos.org/t/video-tutorial-how-to-fix-broken-haskell-packages-in-nix/3968

(Really the only change you need to make is to not edit the hackage-packages.nix file by hand.  It gets regenerated automatically based on this configuration-hackage2nix.yaml file.)

Comment (emilazy) [reply pkgs/development/haskell-modules/configuration-hackage2nix.yaml]: done, thanks for letting me know!

File: pkgs/development/python-modules/nmigen/default.nix [hunk 2]
```diff
@@ -38,25 +33,7 @@ buildPythonPackage rec {
...
-      tool symbiyosys "sby" //
-      tool nextpnr "nextpnr-ice40" //
-      tool nextpnr "nextpnr-ecp5" //
-      tool icestorm "icepack" //
-      tool trellis "ecppack";
-  in ''
-    substituteInPlace nmigen/_toolchain.py \
-      --replace 'overrides = {}' \
-                'overrides = ${builtins.toJSON toolchainOverrides}'
-  '';
```
Comment (thoughtpolice) [inline pkgs/development/python-modules/nmigen/default.nix]: ❤️

**Review by thoughtpolice — APPROVED** [review]
Good stuff as usual, thanks @emilazy. Let's go ahead and merge -- since our aarch64 builders are the fastest and actually completed, I'll assume things are perfect in every way and nothing could possibly go wrong (if something spooky crops up on say, Darwin, we'll see it in Hydra later)

**Pull request merged**